docs/development.rst

```diff
@@ -180,6 +186,64 @@ Please follow `this guide
 <https://stdpopsim.readthedocs.io/en/stable/development.html#rebasing>`_
 for step-by-step instructions on rebasing and squashing commits.
 
+
+.. _sec_development_workflow_anothers_commit:
+
+----------------------------------------
+Checking out someone else's pull request
+----------------------------------------
+
+Sometimes you want to just check out someone else's pull request,
+for the purpose of trying it out and giving them feedback.
+To do this, you first need your own local version of the git repository,
+so you should first do steps 1 and 2 above.
+(Strictly speaking, you don't need a fork on github
+if you con't plan to edit, but it won't hurt.)
+Continuing from there, let's say you want to check out the current
+state of the code on `pull request #854 <https://github.com/tskit-dev/tskit/pull/854>`_.
+(So, below you should replace ``854`` with the number of the pull request
+that you actually want to investigate.)
+Then, continuing from above:
+
+3. Fetch the pull request, and store it as local branch:
+    
+    $ git fetch upstream pull/854/head:my_pr_copy
```

Comment:
The "remote" repository will probably be called `origin` if they just cloned (rather than forking and following github's instructions). The output of `git remote` will show the possible options if one is unsure.